### PR TITLE
feat: All entries page + Navbar

### PR DESCRIPTION
**Pull request by aadarsh-ram**
### What
- Completed an "All entries" page
- Added a Navbar component

### Screenshot
allentries-nav

### CODE REVIEW

File: taxonomy-editor-frontend/src/pages/allentries/EntriesList.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import Paper from '@mui/material/Paper';
+import EditIcon from '@mui/icons-material/Edit';
+import IconButton from '@mui/material/IconButton';
+import { useNavigate } from "react-router-dom";
+
+const EntriesList = ({ nodes, title }) => {
+    const navigate = useNavigate();
+
+    // Handler function for button clicks
+    const handleClick = (event, id) => {
```
Comment (nobeeakon) [inline taxonomy-editor-frontend/src/pages/allentries/EntriesList.jsx:17]: what about using directly a `Link`?

File: taxonomy-editor-frontend/src/pages/allentries/EntriesList.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        event.preventDefault();
+        navigate('/entry/'+id);
+    }
+    return ( 
+        <div className="entry">
```
Comment (nobeeakon) [inline taxonomy-editor-frontend/src/pages/allentries/EntriesList.jsx:22]: please avoid using className, you could try `<Box sx={{}}>`. using hardcoded direct css classes tend to be difficult to manage  at some point :(

File: taxonomy-editor-frontend/src/pages/allentries/index.jsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        return (
+            <div className="all-entries">
+                {isError && <div>{errorMessage}</div>}
+            </div>
+        )
+    }
+    return (
+        <div className="all-entries">
+            {isPending && <div>Loading...</div>}
+            {isSuccess && <EntriesList nodes={nodes} title={"Test"}/>}
```
Comment (nobeeakon) [inline taxonomy-editor-frontend/src/pages/allentries/index.jsx:17]: nit: if there's no plan to include more things in this component, I think the content of `EntriesList` could live here (as is `EntriesList` is mainly `jsx` it should't be much of a problem)

File: taxonomy-editor-frontend/src/components/ResponsiveAppBar.jsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+                width="50px"
+                height="50px"
+                alt="OpenFoodFacts logo"
+              />
+            </MuiLink>
+            <Typography
+              variant="h6"
+              noWrap
+              component="a"
+              href="/"
```
Comment (nobeeakon) [inline taxonomy-editor-frontend/src/components/ResponsiveAppBar.jsx:142]: maybe use `Link` instead?

File: taxonomy-editor-frontend/src/pages/allentries/index.jsx [hunk 5]
```diff
@@ -1,20 +1,83 @@
...
+            <div className="all-entries">
+                {isPending && <div>Loading...</div>}
+            </div>
+        )
+    }
     return (
         <div className="all-entries">
-            {isPending && <div>Loading...</div>}
-            {isSuccess && <EntriesList nodes={nodes} title={"Test"}/>}
+            { isSuccess &&
```
Comment (nobeeakon) [inline taxonomy-editor-frontend/src/pages/allentries/index.jsx:33]: nit: one of the benefits of having an status vs different states in the fetch is that it is unsured that these different statuses (and its derived boolean  counterparts `isSuccess`, `isPending`, ...) are mutually exclusive, in this case as you have already checked that it is not loading or is not error then the only remaining option is `isSuccess`, so in principle there shouldn't be a need to check in here.

Comment (aadarsh-ram) [reply taxonomy-editor-frontend/src/pages/allentries/index.jsx:33]: Oh yeah, that makes sense. I'll remove that check.

**Review by nobeeakon — APPROVED** [review]
:+1:

**Pull request merged**